README.md

```diff
@@ -1,11 +1,43 @@
 # dm-spark-tpcxbb
+
 Spark SQL homework based on TPCx-BB for the Data Management course
 
 ## Get started
-- Read the homework sheet
-- Clone/download this repository
-- Install Jupyter lab and run it: 
-```sh
-pip install jupyterlab
-jupyter lab
+
+- Read the homework sheet delivery requirements
+- Clone this repository
+- Perform the setup steps to get a clean working environment
+
+## Setup
+
+The dependency management of the python3.10 project is done using 
+[poetry](https://python-poetry.org/docs/).
+It will create a virtual environment and install the dependencies in it.
+
+If you prefer to manually install the dependencies, 
+check out the `pyproject.toml`.
+
+Here a script to leverage the automated installation of
+the virtual environment and deps on a Debian-based systems:
+```shell
+# install pipx
+sudo apt update             && \
+sudo apt install -y pipx    && \
+pipx ensurepath             && \
+
+# install poetry
+pipx install poetry         && \
+
+# from repository root where pyproject.toml is located
+poetry install              && \
+
+# to run the shell within the virtual environment
+poetry shell;
 ```
+
+Suggestion: 
+use the same python environment in Pycharm IDE 
+by selecting the python interpreter from the poetry venv[^1] 
+to benefit from its integration.
+
```

Comment:
could you add instructions for starting jupyter? Either `jupyter notebook` or `jupyter lab` depending on what we end up using

Reply:
Added relative run instructions for jupyter lab environment